core/src/main/java/org/apache/iceberg/BaseOverwriteFiles.java

```diff
@@ -163,6 +163,12 @@ protected void validate(TableMetadata base, Snapshot parent) {
     }
   }
 
+  @Override
+  protected boolean useReferencedManifestsWhenFiltering(Snapshot parent) {
```

Comment:
Would something like this be easier to read? Yet to check if that's what we want.

```
@Override
protected boolean useReferencedManifestsWhenFiltering(Snapshot parent) {
  Long parentSnapshotId = parent != null ? parent.snapshotId() : null;
  return Objects.equals(startingSnapshotId, parentSnapshotId);
}
```